sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -148,6 +149,8 @@ private[sql] class SharedState(val sparkContext: SparkContext) extends Logging {
 
 object SharedState {
 
+  URL.setURLStreamHandlerFactory(new SparkUrlStreamHandlerFactory())
```

Comment:
IMHO I think directly register `FsUrlStreamHandlerFactory` with `URL#setURLStreamHandlerFactory` should be enough, it is not necessary to wrap again.